src/Azure.Functions.Cli/Common/Utilities.cs

```diff
@@ -189,36 +182,23 @@ internal static string EnsureCoreToolsLocalData()
             return localPath;
         }
 
-        internal static LogLevel GetHostJsonDefaultLogLevel(IConfigurationRoot hostJsonConfig)
-        {
-            string defaultLogLevelKey = ConfigurationPath.Combine(ConfigurationSectionNames.JobHost, ConfigurationSectionNames.Logging, LogLevelSection, LogLevelDefaultSection);
-            try
-            {
-                if (Enum.TryParse(typeof(LogLevel), hostJsonConfig[defaultLogLevelKey].ToString(), true, out object outLevel))
-                {
-                    return (LogLevel)outLevel;
-                }
-            }
-            catch
-            {
-            }
-            // Default log level
-            return LogLevel.Information;
-        }
-
-        internal static bool LogLevelExists(IConfigurationRoot hostJsonConfig, string category)
+        internal static bool LogLevelExists(IConfigurationRoot hostJsonConfig, string category, out LogLevel outLogLevel)
         {
+            // Host creates user logger for function with Category: Function.{FunctionName}.User Remove ".User" to match the  category name to the funciton name
+            category = category.Replace(".User", "", StringComparison.OrdinalIgnoreCase);
             string categoryKey = ConfigurationPath.Combine(ConfigurationSectionNames.JobHost, ConfigurationSectionNames.Logging, LogLevelSection, category);
             try
             {
                 if (Enum.TryParse(typeof(LogLevel), hostJsonConfig[categoryKey], true, out object outLevel))
```

Comment:
This shouldn't be a direct match -- we need to do a `StartsWith` here -- and choose the longest match. Then you don't need to do the replacement of `.User`. The built-in `LoggerFilterOptions` does all of this already so you shouldn't really need this logic if you switched to using that.

For example -- if someone uses two filters of `Host.` -> `LogLevel.None` and `Host.Startup` -> `LogLevel.Information`. it means they want all host logs to be hidden unless it's `Host.Startup`. I suspect this doesn't work right here?

Reply:
Can we please have a test in place for those scenarios?

Reply:
Updated to rely on `LoggerFilterOptions`. Added unit test that verified rule selection.